### PR TITLE
chore(NODE-5446): update driver v6 dependencies

### PR DESCRIPTION
**Pull request by W-A-James**
#### What is changing?
Update the following dev dependencies
* "@microsoft/api-extractor": "^7.36.4"
* "@octokit/core": "^4.2.4"
* "@types/node": "^20.4.8"
* "@types/sinon": "^10.0.16"
* "@typescript-eslint/eslint-plugin": "^5.62.0"
* "@typescript-eslint/parser": "^5.62.0"
* "eslint": "^8.46.0"
* "eslint-config-prettier": "^8.10.0"
* "eslint-plugin-import": "^2.28.0"
* "gcp-metadata": "^5.3.0"
* "semver": "^7.5.4"
* "sinon": "^15.2.0"
* "v8-heapsnapshot": "^1.3.1"

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

NODE-5446; keeping tooling up to date

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,29 +66,29 @@
...
     "chai": "^4.3.7",
     "chai-subset": "^1.6.0",
     "chalk": "^4.1.2",
-    "eslint": "^8.40.0",
-    "eslint-config-prettier": "^8.8.0",
-    "eslint-plugin-import": "^2.27.5",
+    "eslint": "^8.46.0",
+    "eslint-config-prettier": "^8.10.0",
+    "eslint-plugin-import": "^2.28.0",
     "eslint-plugin-prettier": "^4.2.1",
```
Comment (baileympearson) [inline package.json:92]: did we decide not to bump gcp-metadata?

**Review by baileympearson — CHANGES_REQUESTED** [review]
question: how come we decided to disable `ban-types` in most of our lint configurations?

**Comment by W-A-James** [comment]
@baileympearson re disabling ban-types, when updating the eslint and associated dev dependencies, we start getting the following lint failures

```
/home/wajames/node_driver/src/operations/operation.ts
  20:47  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types

/home/wajames/node_driver/test/types/basic_schema.test-d.ts
  59:14  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.
- If you really want a type meaning "any non-nullish value", you probably want `NonNullable<unknown>` instead  @typescript-eslint/ban-types

/home/wajames/node_driver/test/types/community/changes_from_36.test-d.ts
  16:15  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape                                  @typescript-eslint/ban-types
  24:15  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.
- If you really want a type meaning "any non-nullish value", you probably want `NonNullable<unknown>` instead  @typescript-eslint/ban-types

X 4 problems (4 errors, 0 warnings)
```

I wasn't sure if we wanted to address those here with code changes, so I disabled that rule, which in retrospect might have been a reach too far. I can disable the rule on a per-line basis and file a follow-up ticket.

File: .eslintrc.json [hunk 2]
```diff
@@ -242,6 +243,7 @@
         "@typescript-eslint/no-unsafe-assignment": "off",
         "@typescript-eslint/no-unsafe-return": "off",
         "@typescript-eslint/no-unsafe-call": "off",
+        "@typescript-eslint/ban-types": "off",
```
Comment (nbbeeken) [inline .eslintrc.json:246]: Let's fix the one violation in src: we can replace OperationConstructor with:
```ts
 { new (...args: any[]): any }
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

**Review by nbbeeken — APPROVED** [review]

**Review by baileympearson — APPROVED** [review]

**Pull request merged**